force-app/main/default/classes/ContactRollupsSchedulable.cls

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ *  * Copyright (c) 2021, salesforce.com, inc.
+ *  * All rights reserved.
+ *  * SPDX-License-Identifier: BSD-3-Clause
+ *  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ *
+ */
+
+public class ContactRollupsSchedulable implements Database.Batchable<SObject>, Schedulable {
+    @TestVisible
+    private static AttendanceService attendanceService = new AttendanceService();
+
+    public void execute(SchedulableContext context) {
+        Database.executeBatch(new ContactRollupsSchedulable());
+    }
+
+    public Database.Querylocator start(Database.BatchableContext bc) {
+        List<String> attendanceStatuses = attendanceService.getAttendanceStatuses();
+
+        QueryBuilder query = new QueryBuilder();
+        query.withSObjectType(Contact.SObjectType);
+        query.withSelectFields(
+            new List<String>{
+                String.valueOf(Contact.NumPresentServiceDeliveries__c),
+                String.valueOf(Contact.NumAbsentServiceDeliveries__c),
+                String.valueOf(Contact.ConsecutiveAbsences__c),
+                String.valueOf(Contact.LastServiceDate__c)
+            }
+        );
+
+        QueryBuilder subQuery = new QueryBuilder();
+        subQuery.withChildRelationshipName('Service_Deliveries__r');
```

Comment:
I didn't catch this in the last review but should we be getting the relationship name here dynamically?